src/config.js

```diff
@@ -1168,15 +1211,22 @@ class Config {
   */
 
   priorityForSource (source) {
-    return (source === this.mainSource) ? 1000 : 0
+    switch (source) {
+      case this.mainSource:
+        return 1000
+      case this.projectFile:
+        return 2000
+      default:
+        return 0
+    }
   }
 
   emitChangeEvent () {
     if (this.transactDepth <= 0) { return this.emitter.emit('did-change') }
   }
 
-  resetUserScopedSettings (newScopedSettings) {
-    const source = this.mainSource
+  resetUserScopedSettings (newScopedSettings, options = {}) {
+    const source = options.source == null ? this.mainSource : options.source
```

Comment:
Maybe we should rename this to `resetScopedSettings` since it's not necessarily the user's settings any more.